packages/stream-metadata/src/riverStreamRpcClient.ts

```diff
@@ -149,6 +155,38 @@ function stripHexPrefix(hexString: string): string {
 	return hexString
 }
 
+const workerPath = path.join(__dirname, 'unpackStreamWorker.cjs')
```

Comment:
@brianathere we should avoid source references to build if we can

Reply:
This is how nodejs workers are loaded. If you have another way to make the source available I'm interested in learning.

Reply:
from [nodejs docs](https://nodejs.org/api/worker_threads.html). They put the main thread code and the worker code in the same file:

```js
const {
  Worker, isMainThread, parentPort, workerData,
} = require('node:worker_threads');

if (isMainThread) {
  module.exports = function parseJSAsync(script) {
    return new Promise((resolve, reject) => {
      const worker = new Worker(__filename, {
        workerData: script,
      });
      worker.on('message', resolve);
      worker.on('error', reject);
      worker.on('exit', (code) => {
        if (code !== 0)
          reject(new Error(`Worker stopped with exit code ${code}`));
      });
    });
  };
} else {
  const { parse } = require('some-js-parsing-library');
  const script = workerData;
  parentPort.postMessage(parse(script));
}
```

We could do something along these lines:

```ts
import {
  Worker, isMainThread, parentPort, workerData,
} from 'node:worker_threads';

export function parseJSAsync(script) {
    return new Promise((resolve, reject) => {
      const worker = new Worker(__filename, {
        workerData: script,
      });
      worker.on('message', resolve);
      worker.on('error', reject);
      worker.on('exit', (code) => {
        if (code !== 0)
          reject(new Error(`Worker stopped with exit code ${code}`));
      });
    });
  };

if (!isMainThread) {
  const { parse } = require('some-js-parsing-library');
  const script = workerData;
  parentPort.postMessage(parse(script));
}
```

Reply:
something else i saw while reading the docs:

> The above example spawns a Worker thread for each parseJSAsync() call. In practice, use a pool of Workers for these kinds of tasks. Otherwise, the overhead of creating Workers would likely exceed their benefit.

In our case there will always be a benefit since we avoid blocking the http server, with or without pools. But pooling does sound like something we could benefit from